lib/puppet/type/file.rb

```diff
@@ -881,6 +885,9 @@ def write(property = nil)
 
         fail_if_checksum_is_wrong(file.path, content_checksum) if validate_checksum?
         if self[:validate_cmd]
+          # verify setting windows perms
+          file.chmod(mode_int)
+          file.chown(uid, gid) if Puppet.features.root?
```

Comment:
@joshcooper this seems to do the trick to only manage ownership if running as root. Should we add a warning message or anything to let users know?

Reply:
Ah yeah. The file type detects if the desired owner/group is different than the current value and warns if we're not root:

```ruby
      return true if @should.include?(current)

      unless Puppet.features.root?
        warnonce "Cannot manage ownership unless running as root"
        return true
      end
```

That's suprising to me actually. I would think it would be a hard error, or it would warn and then call `chown` (and fail).

Reply:
I think the `chown` here isn't defined in puppet anywhere. I'm fairly sure it comes from the ruby file object. I'm not sure if you're telling me that I can remove this feature check, or if you're giving additional information around what's happening here?

Reply:
oh sorry, I meant the class method `File.chown`.